### PR TITLE
Implement GenAI Response Guardrail

### PR DESCRIPTION
**Pull request by Monica-Coira**
## Description

- Implements a response guardrail for the study goals generated by the Gemini 2.5 Flash model by calling the model to check for possible biases, derogatory information, factual correctness, incorrect class content, and incorrect formatting.
- If the model finds any of these issues, the response is rewritten and the guardrail is called again until no more issues are found or until the guardrail has been called 3 times.
- In the next PR, I will work on the stretch feature that allows users to delete a group.

## Milestones
This works towards Milestone 15, which is a GenAI Response Guardrail.

## Resources
How to Use Guardrails to Design Safe and Trustworthy AI: [Guardrails](https://medium.com/@caldhubaib/how-to-use-guardrails-to-design-safe-and-trustworthy-ai-c4574eb99243)

## Test Plan

https://github.com/user-attachments/assets/233647fd-c3ba-4096-b72e-4b58e44d1b66

Besides testing basic functionality, these are the corner cases I tested:

- Ensured that the guardrail will not be called more than three times for the same response to prevent reaching the token usage limit of the Gemini API.
- Ensured that if the response meets all the requirements, the guardrail is no longer called and the most updated response is displayed on the screen.
- Ensured that only responses that have been passed through the guardrail are displayed and saved in the database.

### CODE REVIEW

File: client/src/components/GenAIGuardrail.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { GEMINI_API_KEY } from "../utils/apiConfig";
+
+const GenAIGuardrail = async (generatedResponse) => {
+    const ai = new GoogleGenAI({apiKey: GEMINI_API_KEY});
+    let checkCount = 0;
+    let finalResponse;
+    while (checkCount <= 3){
+        const checkedResponse = await ai.models.generateContent({
+            model: "gemini-2.5-flash",
+            contents: `You are a guardrail bot and have been given this response: ${generatedResponse}.
```
[the remark this answers is not in the document]
Comment (Monica-Coira) [reply client/src/components/GenAIGuardrail.jsx:11]: Got it, just refactored the code to store the prompt contents for the goals generator and response guardrail in the apiConfig file in PR #56 .

File: client/src/components/GenAIGuardrail.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            model: "gemini-2.5-flash",
+            contents: `You are a guardrail bot and have been given this response: ${generatedResponse}. 
+            Check for the following issues, and if any issues are found, fix the issues and rewrite the response.
+            Is the response in a list format with 5 different study goals? Does the response not contain any harmful biases or derogatory information?
+            Is the tone safe and appropriate for college students? Is the result factually correct? Do all five study goals relate to the same class?
+            Does a college student have the right to access and know all of the information in the response? 
+            If the answer to any of these questions is no, rewrite the response in the same format while fixing the issues. 
+            If the response to all of these questions is yes, output only the word "yes" without the response.`,
+        });
+        if (checkedResponse.text === "yes"){
```
[the remark this answers is not in the document]
Comment (Monica-Coira) [reply client/src/components/GenAIGuardrail.jsx:19]: In the guardrail prompt, I instruct the model to only output the word "yes" without the response if the response passes all of the checks. I tested this to ensure that the model always outputs "yes" in the end, even if the original response doesn't pass the checks, and found that all responses have passed the checks after the first iteration.
